src/status_im/utils/views.clj

```diff
@@ -66,3 +66,210 @@
+                     :current-view ~view-sym}])
+                  grandchildren)]))
+
+(defn -compile-views [n views]
+  (let [view-sym (gensym "view-id")]
+    `(defview ~n []
+       (letsubs [~view-sym [:get :view-id]]
+        ~(let [tree (-build-tree views)]
+           (-generate-component tree view-sym :root))))))
+
+(defmacro compile-views
+  [n views]
+  (-compile-views n views))
+
+
+(comment
+
+ (-compile-views
+  'root-view
+  '[{:views     #{:home :wallet :my-profile}
+     :component main-tabs}
+
+    {:view      :chat
+     :hide?     (not android?)
+     :component chat}
+
+    {:view      :wallet-send-transaction
+     :parent    :wallet
+     :component send-transaction}
+
+    {:view      :wallet-request-transaction
+     :parent    :wallet
+     :component request-transaction}
+
+    {:view      :choose-recipient
+     :parent    :wallet-send-transaction
+     :component choose-recipient}
+
+    {:view      :wallet-transaction-sent
+     :parent    :wallet-send-transaction
+     :component transaction-sent}
+
+    {:views     #{:transactions-history :unsigned-transactions}
+     :parent    :wallet
+     :component wallet-transactions/transactions}
+
+    {:view      :edit-my-profile
+     :parent    :profile
+     :component edit-my-profile}])
+
+ (status-im.utils.views/defview
+   root-view
+   []
+   (status-im.utils.views/letsubs
+     [view-id76826 [:get :view-id]]
+     [status-im.ui.components.react/view
+      {:flex 1}
+      [status-im.ui.components.react/navigation-wrapper
+       {:views        #{:home :wallet :my-profile},
+        :component    main-tabs,
+        :hide?        nil,
+        :current-view view-id76826}]
+      [status-im.ui.components.react/navigation-wrapper
+       {:views        #{:chat},
+        :component    chat,
+        :hide?        (not android?),
+        :current-view view-id76826}]
+      [status-im.ui.components.react/navigation-wrapper
+       {:hide?        true,
+        :component    [status-im.ui.components.react/with-empty-preview
+                       [status-im.ui.components.react/view
+                        {:flex (if
+                                (clojure.core/contains?
+                                 #{:wallet-request-transaction
+                                   :wallet-send-transaction
+                                   :choose-recipient
+                                   :wallet-transaction-sent
+                                   :transactions-history
+                                   :unsigned-transactions}
+                                 view-id76826)
+                                 1
+                                 0)}
+                        [status-im.ui.components.react/navigation-wrapper
+                         {:views        #{:wallet-send-transaction},
+                          :component    send-transaction,
+                          :hide?        nil,
+                          :current-view view-id76826}]
```

Comment:
What is this? Seems like a manual `gensym`

Reply:
this is a comment which shows how it expands. And yes this symbol was generated by gensym :)

Reply:
duh, missed parent form, makes more sense :D